### PR TITLE
ES2018 implementation

### PR DESCRIPTION
**Pull request by Protryon**
Implemented [es2018](https://www.ecma-international.org/ecma-262/9.0/index.html) spec.

### CODE REVIEW

**Review by michaelficarra — APPROVED** [review]

**Comment by bakkot** [comment]
We need to add the [`s` flag](https://github.com/tc39/ecma262/pull/1028) to the possible flags for `LiteralRegExpExpression`, yes?

Otherwise LGTM. I wasn't sure exactly which commits got in, so I compared [4b21985](https://github.com/tc39/ecma262/commit/4b2198595da6599ad6a9390c8225646803faf3a2) to [8340bf9](https://github.com/tc39/ecma262/commit/8340bf9a8427ea81bb0d1459471afbcc91d18add).

**Comment by Protryon** [comment]
@bakkot Updated.

**Comment by bakkot** [comment]
I was going to say that `dotAll` should come after `multiline`, but I just noticed that we're already deviating from the [spec order](https://tc39.github.io/ecma262/#sec-get-regexp.prototype.flags) in putting `y` before `u`. Don't know what to do about that. Otherwise LGTM.

No idea why CI's failing, but that'll need to get resolved before this can land.

**Comment by Protryon** [comment]
@bakkot CI fixed, I have no idea how it ever passed, `npm i` was not called.

**Comment by michaelficarra** [comment]
Let's change the flag order to match spec order and follow that from here on.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 sudo: false
 script:
   - curl https://raw.githubusercontent.com/shapesecurity/CLA/master/cla-travis.sh | bash
+  - npm i
```
Comment (michaelficarra) [inline .travis.yml]: `npm i` is a nice shorthand when using the CLI. When writing down commands in scripts, use long flag names and full subcommand names. `npm install`.

**Comment by Protryon** [comment]
@michaelficarra Updated travis to use `npm install`, reflected spec in `LiteralRegExpExpression`.

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**